src/main/kotlin/no/nav/familie/ks/sak/kjerne/beregning/TilkjentYtelseValidator.kt

```diff
@@ -32,27 +30,35 @@ object TilkjentYtelseValidator {
     fun validerAtTilkjentYtelseHarFornuftigePerioderOgBeløp(
         tilkjentYtelse: TilkjentYtelse,
         personopplysningGrunnlag: PersonopplysningGrunnlag,
-        vilkårsvurdering: Vilkårsvurdering,
+        behandlingSkalFølgeNyeLovendringer2024: Boolean,
     ) {
         val søker = personopplysningGrunnlag.søker
         val barna = personopplysningGrunnlag.barna
 
+        val andelerTilkjentYtelse = tilkjentYtelse.andelerTilkjentYtelse.toList()
+
+        if (hentSøkersAndeler(andeler = andelerTilkjentYtelse, søker).isNotEmpty()) {
+            throw Feil("Feil i beregning. Søkers andeler må være tom")
+        }
+
         val andelerPerAktør = tilkjentYtelse.andelerTilkjentYtelse.groupBy { it.aktør }
 
         andelerPerAktør.filter { it.value.isNotEmpty() }.forEach { (aktør, andeler) ->
-            val regelsettPåAktør = vilkårsvurdering.hentPersonResultaterTilAktør(aktørId = aktør.aktørId).hentRegelsettBruktIVilkår() ?: throw Feil("Fant ikke regelsett på aktør med id ${aktør.aktørId}")
-
-            val maksAntallMånederMedUtbetaling =
-                when (regelsettPåAktør) {
-                    VilkårRegelsett.LOV_AUGUST_2021 -> 11
-                    VilkårRegelsett.LOV_AUGUST_2024 -> 7
-                }
 
             val stønadFom = andeler.minOf { it.stønadFom }
             val stønadTom = andeler.maxOf { it.stønadTom }
 
+            val relevantBarn = barna.single { it.aktør == aktør }
+
+            val vilkårLovverkInformasjonForBarn = VilkårLovverkInformasjonForBarn(relevantBarn.fødselsdato)
+
+            val maksAntallMånederMedUtbetaling =
+                when (behandlingSkalFølgeNyeLovendringer2024) {
+                    true -> utledMaksAntallMånederMedUtbetaling(vilkårLovverkInformasjonForBarn)
+                    false -> 11L
+                }
             val diff = Period.between(stønadFom.toLocalDate(), stønadTom.toLocalDate())
-            if (diff.toTotalMonths() > maksAntallMånederMedUtbetaling) {
+            if (diff.toTotalMonths() >= maksAntallMånederMedUtbetaling) {
```

Comment:
Denne tror jeg blir feil. Skal vel være mulig å ha likt antall måneder som maks?

Reply:
Enig, det skal være mulig å ha likt antall måneder som maks!

Reply:
Ser igjen her at det er diff.totalMonths som brukes, altså diff på måneder mellom to datoer.
Dersom fom og tom er 2024-04 som er 1 måned med utbetaling vil totalMonths være 0, så det er riktig med =

Reply:
Hmm ikkesant. Da lurer jeg på om jeg syns det er med lesbart å plusse på 1 på totalmonths og kun sjekke på større enn🤔